DPPP/test/unit/tBDABuffer.cc

```diff
@@ -0,0 +1,391 @@
+// tScaleData.cc: Test program for class ScaleData
+// Copyright (C) 2020
+// ASTRON (Netherlands Institute for Radio Astronomy)
+// P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
+//
+// This file is part of the LOFAR software suite.
+// The LOFAR software suite is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The LOFAR software suite is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
+//
+// @author Lars Krombeen
+
+#include "../Common/Types.h"
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/data/test_case.hpp>
+
+#include "../../BDABuffer.h"
+
+using DP3::DPPP::BDABuffer;
+
+BOOST_AUTO_TEST_SUITE(bdabuffer)
+
+BOOST_AUTO_TEST_CASE( test_bda_initialization )
+{
+    BDABuffer buffer {2};
+    BOOST_CHECK_EQUAL(buffer.GetNumberOfElements(), size_t {0});
+    BOOST_CHECK_EQUAL(buffer.GetData(), nullptr);
+    BOOST_CHECK_EQUAL(buffer.GetFlags(), nullptr);
+    BOOST_CHECK_EQUAL(buffer.GetWeights(), nullptr);
+    BOOST_CHECK_EQUAL(buffer.GetFullResFlags(), nullptr);
+}
+
+BOOST_AUTO_TEST_CASE( test_copy )
+{
+    const size_t n_channels {2};
+    const size_t n_correlations {3};
+    BDABuffer buffer {3 * n_channels * n_correlations + 1};
+    bool flags[] {true};
+    const std::complex<float> data {1};
+    float weights[] {1};
+
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, &data);
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, flags);
+    buffer.AddRow(0., 1., 0, 0, n_channels, n_correlations, nullptr, nullptr, weights);
+
+    BDABuffer buffer_copy {buffer};
+
+    BOOST_CHECK(buffer.GetData() != buffer_copy.GetData());
+    BOOST_CHECK_EQUAL(*buffer.GetData(), *buffer_copy.GetData());
```

Comment:
You're only comparing the first data element here. Please augment this test:
- Test that the number of elements in the copy is correct.
- Test that all individual elements are equal in both buffers.
- Test that you cannot add any rows to the copy.